structuralcodes/codes/mc2020/_concrete_material_properties.py

```diff
@@ -0,0 +1,2833 @@
+"""fib MC2020 Chapter 14.6.1."""
+
+import math
+from typing import List, Literal, Optional, Tuple
+
+
+def fcm(fck: float, delta_f: float = 8) -> float:
+    """Calculate the mean compressive strength fcm based on the
+        characteristic compressive strength fck.
+
+    fib Model Code 2020, eq. (14.6-1)
+
+    Args:
+        fck (float): Characteristic compressive strength fck in MPa.
+        delta_f (float): increment in MPa. Defaults to 8MPa.
+
+    Returns:
+        float: Mean compressive strength fcm in MPa.
+
+    Raises:
+        ValueError: If fck is negative or zero.
+    """
+    if fck <= 0:
+        raise ValueError(
+            'Characteristic compressive strength fck must be positive.'
+        )
+
+    return fck + delta_f
+
+
+def flcm(flck: float, delta_f: float = 8) -> float:
+    """Calculate the mean compressive strength flcm for lightweight
+        aggregate concrete based on the characteristic
+        compressive strength flck.
+
+    fib Model Code 2020, eq. (14.6-2)
+
+    Args:
+        flck (float): Characteristic compressive strength flck in MPa.
+
+    Returns:
+        float: Mean compressive strength flcm in MPa.
+        delta_f (float): increment in MPa. Defaults to 8MPa.
+
+    Raises:
+        ValueError: If flck is negative or zero.
+    """
+    if flck <= 0:
+        raise ValueError(
+            'Characteristic compressive strength flck must be positive.'
+        )
+
+    return flck + delta_f
+
+
+# Dictionary for normal concrete grades and
+# their respective fck and fck,cube values
+_concrete_strengths = {
```

Comment:
I think this is a clever solution when the data is given in a table in the code, but it might be a short-coming when we start implementing methods on the material level, and the characteristic cylinder strength is the expected input and not the strength class. @talledodiego any thoughts on this?